### PR TITLE
feat: bypass cache on screenshots for alerts

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

When creating a screenshot for an alert we should bypass the cache. This PR changes the alert jobs to pass `force=true` when fetching a screenshot, and also implements the logic to carry the `force` parameter through the stack.

### TESTING INSTRUCTIONS

I added unit tests to the backend and frontend.

To test locally:

1. Set up an alert that triggers (eg, `SELECT COUNT(*) FROM bart_lines` and `>0`).
2. Check that the celery worker takes a screenshot with `?force=true` (you can see it in the logs).
3. Click the link, the image should open fullscreen and bypassing the cache.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
@betodealmeida does this use a different cache key for reports or by running an alert/report that busts the cache, does it update the explore view as well?

**Comment by betodealmeida** [comment]
> @betodealmeida does this use a different cache key for reports or by running an alert/report that busts the cache, does it update the explore view as well?

It's the same cache, so it should update the explore view as well.

File: superset-frontend/src/explore/exploreUtils/getExploreLongUrl.test.ts [hunk 1]
```diff
@@ -45,12 +45,27 @@ test('Get url when endpointType:standalone', () => {
   expect(
     getExploreLongUrl(
       params.formData,
-      params.endpointType,
```
Comment (betodealmeida) [inline superset-frontend/src/explore/exploreUtils/getExploreLongUrl.test.ts:48]: This test was wrong.

**Comment by graceguo-supercat** [comment]
@betodealmeida Thanks for the PR. I have 1 question, this change only impact `Alert` but not `Report`, right?
Thinking about if a user setup email **_report_** for a large dashboard, and with 1 min as frequency, it might impact  backend query engine.

**Comment by betodealmeida** [comment]
> @betodealmeida Thanks for the PR. I have 1 question, this change only impact `Alert` but not `Report`, right? Thinking about if a user setup email **_report_** for a large dashboard, and with 1 min as frequency, it might impact backend query engine.

Right, this is just for alerts. For reports I have another PR that makes it configurable, with the default being to use the cache.

**Comment by graceguo-supercat** [comment]
@betodealmeida If alert triggers a dashboard, will every chart in the dashboard have force refresh?

**Comment by betodealmeida** [comment]
> @betodealmeida If alert triggers a dashboard, will every chart in the dashboard have force refresh?

That's a great question... right now I'm adding this just for charts, not dashboards. Maybe for dashboards we should also make it configurable (default off), since the impact can be much bigger?

**Comment by graceguo-supercat** [comment]
> Maybe for dashboards we should also make it configurable (default off), since the impact can be much bigger?

agree. Thanks!

**Review by eschutho — APPROVED** [review]

**Pull request merged**